flaml/nlp/autotransformers.py

```diff
@@ -0,0 +1,787 @@
+                self._test_name]
+
+    def _set_model_config(self, checkpoint_path, per_model_config, model_config_num_labels):
+        if per_model_config and len(per_model_config) > 0:
+            model_config = AutoConfig.from_pretrained(
+                checkpoint_path,
+                num_labels=model_config_num_labels,
+                **per_model_config)
+        else:
+            model_config = AutoConfig.from_pretrained(
+                checkpoint_path,
+                num_labels=model_config_num_labels)
+        return model_config
+
+    def  _load_model(self,
+                    checkpoint_path = None,
+                    per_model_config=None):
+        this_task = get_default_task(self.jobid_config.dat[0], self.jobid_config.subdat)
+        if this_task == "seq-classification":
+            self._num_labels = len(self.train_dataset.features["label"].names)
+        elif this_task == "regression":
+            self._num_labels = 1
+
+        if not checkpoint_path:
+            checkpoint_path = self.jobid_config.pre_full
+        if this_task == "seq-classification":
+            num_labels_old = AutoConfig.from_pretrained(checkpoint_path).num_labels
+            if self.jobid_config.pre in MODEL_CLASSIFICATION_HEAD_MAPPING.keys():
+                model_config_num_labels = num_labels_old
+            else:
+                model_config_num_labels = self._num_labels
+            model_config = self._set_model_config(checkpoint_path, per_model_config, model_config_num_labels)
+
+            if self.jobid_config.pre in MODEL_CLASSIFICATION_HEAD_MAPPING.keys():
+                num_labels_old = AutoConfig.from_pretrained(checkpoint_path).num_labels
+                if per_model_config and len(per_model_config) > 0:
+                    model_config = AutoConfig.from_pretrained(
+                        checkpoint_path,
+                        num_labels = num_labels_old,
+                        **per_model_config)
+                else:
+                    model_config = AutoConfig.from_pretrained(
+                        checkpoint_path,
+                        num_labels = num_labels_old)
+
+                if self._num_labels != num_labels_old:
+                    model_config.num_labels = num_labels_old
+                    this_model = AutoModelForSequenceClassification.from_pretrained(checkpoint_path, config=model_config)
+                    model_config.num_labels = self._num_labels
+                    this_model.num_labels = self._num_labels
+                    this_model.classifier = AutoSeqClassificationHead.from_model_type_and_config(self.jobid_config.pre, model_config)
+                else:
+                    this_model = AutoModelForSequenceClassification.from_pretrained(checkpoint_path, config=model_config)
+            else:
+                this_model = AutoModelForSequenceClassification.from_pretrained(checkpoint_path, config=model_config)
```

Comment:
line 308, 313 and 315 are the same. Can you make it occur only once?

Reply:
it's non trivial due to the dependency on model_config